packages/core/core/src/TransformerRunner.js

```diff
@@ -95,36 +78,28 @@ class TransformerRunner {
   }
 
   async runPipeline(
-    input: TransformerInput,
+    input: Asset,
     pipeline: Array<Transformer>,
     cacheEntry: ?CacheEntry,
-    context: TransformContext
+    previousGenerate: ?GenerateFunc
   ) {
     // Run the first transformer in the pipeline.
     let {
       results,
       connectedFiles,
       generate,
       postProcess
-    } = await this.runTransform(input, pipeline[0], context.generate);
-
-    context.generate = generate;
+    } = await this.runTransform(input, pipeline[0], previousGenerate);
 
-    let assets: Array<Asset> = [];
+    let assets: Array<IAsset> = [];
     for (let result of results) {
-      let asset;
-
-      // If this is the first transformer, create a hash for the asset.
-      if (!context.hash) {
-        asset = await transformerResultToAsset(input, result, context);
-        context.hash = asset.hash;
-      }
+      let asset = input.createChildAsset(result);
```

Comment:
If the result is an Asset, could it just be used instead of creating a child asset? Not sure if it matters.

Reply:
i.e. if `result instanced Asset`? Yeah it could, but by copying the instance we ensure that future mutations don't break anything (i.e. if a transformer returned a constant for some reason, or had its own cache).